src/sinks/aws_s3/sink.rs

```diff
@@ -61,11 +62,34 @@ impl RequestBuilder<(S3PartitionKey, Vec<Event>)> for S3RequestOptions {
         let finalizers = events.take_finalizers();
         let s3_key_prefix = partition_key.key_prefix.clone();
 
+        // Key prefix will already contain topic name since we're extracting it from path and
+        // not the actual file name
+        let topic_name = extract_topic_name(&s3_key_prefix);
```

Comment:
Add a TODO to extract this from the message directly instead of using the existing regex method?

Ditto for the Azure equivalent.

Reply:
Ah good catch, we should already have added it as a field that we're extracting it from the message, so dropping the usage of the regex parsing